rust/connlib/tunnel/src/device_channel/tun_linux.rs

```diff
@@ -350,18 +346,95 @@ impl ioctl::Request<SetTunFlagsPayload> {
     }
 }
 
+/// Back up `/etc/resolve.conf` and then modify it in-place
+async fn configure_resolv_conf(
```

Comment:
I've missed the bulk of the discussion around the feature design of this but at what point do we restore the backup back to what it was? Shouldn't we attempt to do this on exit of the program?